sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/GetSamplingRulesResponse.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.aws.trace;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.auto.value.AutoValue;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+@AutoValue
+@JsonDeserialize(builder = GetSamplingRulesResponse.Builder.class)
```

Comment:
Cribbed pattern for AutoValue + Jackson from https://github.com/artem-zinnatullin/AutoJackson

Reply:
this stuff is a bit crazy, but I guess it works!

Reply:
Since this is the response it will mostly be only deserialized, and you can annotate with `@JsonDeserialize(builder = Whatever.Builder.class)` I think you don't need to annotate anything else with `@JsonProperty("property")` other than the the ones in the `Builder` class, unless we are going to serialize it then we would need to have in both... but we can also change the ObjectMapper

Reply:
For instance if using the ObjectMapper with `setPropertyNamingStrategy(PropertyNamingStrategies.UPPER_CAMEL_CASE)`, SamplingRule can just be 

```
@AutoValue
@JsonDeserialize(builder = AutoValue_SamplingRule.Builder.class)
public abstract class SamplingRule {

  public abstract Map<String, String> attributes();

  public abstract Double fixedRate();

  public abstract String httpMethod();

  public abstract String host();

  public abstract Integer priority();

  public abstract Integer reservoirSize();

  public abstract String resourceArn();

  public abstract String ruleArn();

  public abstract String ruleName();

  public abstract String serviceName();

  public abstract String serviceType();

  public abstract String urlPath();

  public abstract Integer version();

  SamplingRule() {}

  public static Builder builder() {
    return new AutoValue_SamplingRule.Builder();
  }

  @AutoValue.Builder
  @JsonPOJOBuilder(withPrefix = "")
  public abstract static class Builder {
    public abstract Builder attributes(Map<String, String> attributes);

    public abstract Builder fixedRate(Double fixedRate);

    @JsonProperty("HTTPMethod")
    public abstract Builder httpMethod(String httpMethod);

    public abstract Builder host(String host);

    public abstract Builder priority(Integer priority);

    public abstract Builder reservoirSize(Integer reservoirSize);

    @JsonProperty("ResourceARN")
    public abstract Builder resourceArn(String resourceArn);

    @JsonProperty("RuleARN")
    public abstract Builder ruleArn(String ruleArn);

    public abstract Builder ruleName(String ruleName);

    public abstract Builder serviceName(String serviceName);

    public abstract Builder serviceType(String serviceType);

    @JsonProperty("URLPath")
    public abstract Builder urlPath(String urlPath);

    public abstract Builder version(Integer version);

    public abstract SamplingRule build();
  }
}
```

Reply:
Also should we have package private constructors in the auto value?